src/classes/OPP_PrimaryContactRoleMerge_TEST.cls

```diff
@@ -0,0 +1,426 @@
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Opportunity Contact Role
+* @description Test Class for the deletion of the duplicate primary OCRs.
+*/
+
+@isTest
+private class OPP_PrimaryContactRoleMerge_TEST {
+
+    /*********************************************************************************************************
+    * @description Given an Opportunity with blank Primary Contact AND the Opportunity has duplicate primary 
+    * OCRs with Role != NPSP Settings Role, when the batch runs, then an error log created, containing a link 
+    * to the Opportunity.
+    */
+    @isTest static void test_LogErrorIfNoOCRsRoleDefinedInNPSPSettings() {
+        String nonNPSPRole = 'NON_NPSP_ROLE';
+
+        deactivateTriggers();
+
+        Account account = new Account(Name = 'TEST_ACCOUNT');
+        insert account;
+        Opportunity opportunity = new Opportunity(
+            Name = 'TEST_OPPORTUNITY',
+            StageName = UTIL_UnitTestData_TEST.getOpenStage(),
+            CloseDate = Date.today(),
+            AccountId = account.Id,
+            Primary_Contact__c = null
+        );
+        insert opportunity;
+
+        Contact contact = new Contact(LastName = 'TEST_CONTACT');
+        insert contact;
+
+        List<OpportunityContactRole> primaryOCRs = new List<OpportunityContactRole>{
+            new OpportunityContactRole(
+                OpportunityId = opportunity.Id,
+                ContactId = contact.Id,
+                IsPrimary = true,
+                Role = nonNPSPRole
+            ),
+            new OpportunityContactRole(
+                OpportunityId = opportunity.Id,
+                ContactId = contact.Id,
+                IsPrimary = true,
+                Role = nonNPSPRole
+            )
+        };
+        insert primaryOCRs;
+
+        activateTriggers();
+        
+        Test.startTest();
+        Database.executeBatch(new OPP_PrimaryContactRoleMerge_BATCH());
+        Test.stopTest();
+
+        List<Opportunity> expectedOpportunities = [
```

Comment:
Could this be extracted into a separate method?

Reply:
I think since it is only 1 line "Database.executeBatch(new OPP_PrimaryContactRoleMerge_BATCH());" we don't need to create another method.

Reply:
I was referring to the query on the expectedOpportunities, I noticed that was repeated throughout your test methods.